includes/Media.php

```diff
@@ -234,6 +248,19 @@ public static function wp_prepare_attachment_for_js( $response, $attachment ) {
 		return $response;
 	}
 
+	/**
+	 * Get poster image data.
+	 *
+	 * @param int $thumbnail_id Attachment ID.
+	 *
+	 * @return array
+	 */
+	public static function get_thumbnail_data( $thumbnail_id ) {
+		$img_src                       = wp_get_attachment_image_src( $thumbnail_id, 'full' );
+		list ( $src, $width, $height ) = $img_src;
+		$generated                     = (bool) get_post_meta( $thumbnail_id, self::POSTER_POST_META_KEY, true );
```

Comment:
Hmm, shouldn't this always use the automatically generated image's width and height?
Otherwise, if the poster is changed from Media Library, the width and height would potentially be incorrect. Thoughts?

Reply:
This generated field is there just in case the feature image (poster) is set in the CMS by the user. This could be for an old video or some other reason. If generated is false, we know not to trust the sizes.

Reply:
But what if the poster was already generated, and then later the user goes and changes the poster -- this means that `generated` would be `true`, however, the `width` and `height` would be from the custom poster's `wp_get_attachment_image_src( $thumbnail_id, 'full' );`.

Reply:
> But what if the poster was already generated, and then later the user goes and changes the poster -- this means that `generated` would be true

Manually set poster images don't have that meta field, no?

Reply:
Ah, my bad, I thought the meta was gotten for the video, not the thumbnail :)

Anyway, the original thought was that if there's an automatically generated poster then perhaps we should store that data and use that one since otherwise, if there's a custom poster, still the incorrect width/height would be shown.

Reply:
The flag to note that is a generated image is on the image attachment, not the video attachment. If the user changes the featured image of the video, then the new image would not have this piece of post meta and get a false value here. As the generated image can not be seen in the CMS, it is impossible to assign a generated image to a different video.

Reply:
<img width="230" alt="Screenshot 2020-04-09 at 10 56 16" src="https://user-images.githubusercontent.com/3294597/78877203-c4720300-7a50-11ea-889e-b76e0f649903.png">

To reproduce:
1. Upload a video to a Story
2. Go to Media Library and change its poster
3. Use that video in a Story

The incorrect width/heigh is used (since the poster is manually chosen).

Reply:
Good catch. Fixed in https://github.com/google/web-stories-wp/pull/1102/commits/1b114a89de1fe814b043a83ce080ecacc07fa275

Reply:
@spacedmonkey Not sure how that change would fix the issue.

For fixing this issue, we would probably need to store the automatically created poster image's ID as meta for the video -- so that no matter the custom poster assigned meanwhile, we'd always use only the automatically generated poster image as the source of truth.

WDYT?

Reply:
@miina Do you mean what if users set the wrong poster in the media library itself? Not in the editor?

Reply:
Correct, if the users set a custom poster in the media library, in that case.

Reply:
I think that is something we can definitely live with for now.

Reply:
Alright. That's something that can't be fixed for existing images with custom poster later, however, that's quite an edge case anyway.